website/pages/docs/migrating-from-express-graphql.mdx

```diff
@@ -0,0 +1,186 @@
+
+- Use a plugin that serves an interactive GraphQL UI alongside your endpoint
+- Or serve a static HTML page that loads your preferred IDE from a CDN
+
+In either case, make sure to restrict access in production environments.
+
+### Step 6: Test your setup
+
+After migrating, verify that your server responds correctly:
+
+- Send queries and mutations using your preferred client.
+- Check for proper HTTP status codes and response shapes.
+- If you're using streaming features like `@defer`, ensure your client can handle `multipart/mixed` responses.
+
+## Best practices
+
+When migrating from `express-graphql` to `graphql-http`, there are a few key differences and potential pitfalls to keep in mind. These tips can help you avoid common issues and ensure a smoother transition.
+
+### Be aware of different error behavior
+
+`graphql-http` follows the GraphQL over HTTP spec closely, which means error formatting and status codes may differ from what you're used to with `express-graphql`. For example:
+
+- Invalid queries may return a `400 Bad Request` instead of a `200 OK`.
+- Errors in parsing or validation are surfaced earlier and more strictly.
+- You can customize error output using the `formatError` option, but it must conform to the spec.
+
+This can affect client expectations if they were relying on `express-graphql`'s more lenient defaults.
+
+### Consider IDE support in development
+
+`express-graphql` includes GraphiQL by default in development mode. `graphql-http` does not.
+
+To restore this functionality, consider:
+
+- Use a middleware plugin that serves an in-browser IDE.
+- Host a static IDE page at a separate endpoint.
+- Ensure the IDE is only available in non-production environments.
+
+### Watch for framework-specific middleware behavior
+
+Since `graphql-http` is framework-agnostic, it does not handle things like body parsing, CORS, or compression. You'll need to ensure those are handled appropriately by your Express setup:
+
+```js
+import cors from 'cors';
+import express from 'express';
+
+app.use(cors());
+app.use(express.json());
+```
+
+This gives you more control but requires a bit more setup.
+
+### Understand streaming and file upload limitations
+
+`graphql-http` supports streaming via `multipart/mixed`, which enables features like `@defer`. However:
+
+- Many clients don't yet support this content type.
+- File uploads are not built-in, and must be handled outside the core handler.
+
+Test your client integration carefully if you rely on these features.
+
+## What's next
+
+Once you're up and running with `graphql-http`, you can start extending your server with validation rules, monitoring, custom error handling, or alternative transports. For more advanced use cases, consider combining `graphql-http` with tools like [`envelop`](https://the-guild.dev/graphql/envelop) or [`graphql-yoga`](https://the-guild.dev/graphql/yoga-server).
```

Comment:
Envelop + graphql-http certainly makes sense, but I think yoga would replace graphql-http rather than be combined with it, although I am not sure if there are more advanced setups that you had in mind. Maybe just for subscriptions?

What the graphql-http project does is suggest a group of compliant servers, we could link to that list:  https://github.com/graphql/graphql-http?tab=readme-ov-file#servers